src/main/java/oidc/endpoints/AuthorizationEndpoint.java

```diff
@@ -64,27 +63,26 @@
 import java.net.URI;
 import java.net.URISyntaxException;
 import java.net.URLDecoder;
-import java.security.NoSuchAlgorithmException;
-import java.security.NoSuchProviderException;
 import java.security.cert.CertificateException;
 import java.util.Arrays;
-import java.util.Collection;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
+import java.util.Set;
 import java.util.stream.Collectors;
 
+import static java.util.stream.Collectors.joining;
 import static java.util.stream.Collectors.toList;
-import static java.util.stream.Collectors.toMap;
+import static java.util.stream.Collectors.toSet;
 
 @Controller
 public class AuthorizationEndpoint implements OidcEndpoint {
 
     private static final Log LOG = LogFactory.getLog(AuthorizationEndpoint.class);
-    private static final List<String> forFreeOpenIDScopes = Arrays.asList("profile", "email", "address", "phone");
+    private static final List<String> forFreeOpenIDScopes = Arrays.asList("openid", "profile", "email", "address", "phone");
 
```

Comment:
Is openid obligatory for client credentials clients now? Or can you just use these at will?

Reply:
First question: no. Second question: yes. These are the scopes that can be asked for and will be granted regardless of what is configured in manage.